shtdlib.sh

```diff
@@ -706,7 +706,7 @@ function priv_esc_with_env {
 function get_custom_ssh_auth_agent {
     docker_ssh_auth_socket_path="${1:-${HOME}/docker-ssh-agent}"
     docker_ssh_auth_pid_file="${2:-${HOME}/.docker-ssh-agent.pid}"
-    if [ -S "${docker_ssh_auth_socket_path}" ] ; then
+    if [ -S "${docker_ssh_auth_socket_path}" ] && pgrep --pidfile ${docker_ssh_auth_pid_file} 2> /dev/null ; then
```

Comment:
On mac the pidfile param is `-F`; errors with `--pidfile`. 

Also can we add a step to remove anything at the socket path before attempting to create the new socket? Else it complains about "Address already in use"

Reply:
Sure, at this point we're kind of fixing ssh-agent bugs but that's what the lib is for :)